sqlglot/generator.py

```diff
@@ -3022,6 +3022,10 @@ def altercolumn_sql(self, expression: exp.AlterColumn) -> str:
         if comment:
             return f"ALTER COLUMN {this} COMMENT {comment}"
 
+        allow_null = self.sql(expression, "allow_null")
+        if allow_null:
+            nullability = "SET" if allow_null == "NOT NULL" else "DROP"
+            return f"ALTER COLUMN {this} {nullability} NOT NULL"
```

Comment:
What's the point of storing a `exp.NotNullColumnConstraint` in this case, if we're not gonna use the SQL it generates into? I think storing a boolean flag instead will be cleaner, e.g. no string comparison will be required.